From: Willem Toorop <willem@nlnetlabs.nl>
Date: Tue, 28 Jan 2014 11:06:38 +0100
Subject: bugfix #548: Double free for answers > 4096

in ldns_resolver_send_pkt.  Thank you sunthought@gmail.com
---
 resolver.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/resolver.c b/resolver.c
index 16efc90..1474dc8 100644
--- a/resolver.c
+++ b/resolver.c
@@ -1117,6 +1117,7 @@ ldns_resolver_send_pkt(ldns_pkt **answer, ldns_resolver *r,
 					ldns_pkt_set_edns_udp_size(query_pkt
 							, 4096);
 					ldns_pkt_free(answer_pkt);
+					answer_pkt = NULL;
 					/* Nameservers should not become 
 					 * unreachable because fragments are
 					 * dropped (network error). We might
